### DISCUSSION TITLE
Get the error  Prisma Client could not find its `schema.prisma` with Serverless, Turborepo PNPM Workspace

### DISCUSSION THREAD

**Discussion by tranthaison1231** [prisma/prisma#21846] [Q&A]
### Bug description

I got the error on AWS cloudwatch
```
"Prisma Client could not find its `schema.prisma`. This is likely caused by a bundling step, which leads to `schema.prisma` not being copied near the resulting bundle. We would appreciate if you could take the time to share some information with us.\nPlease help us by answering a few questions: https://pris.ly/bundler-investigation-error" 
```
when using project with prisma, serverless, turborepo, pnpm workspace
This is my repo: https://github.com/tranthaison1231/chinese-chess

### Prisma information

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "rhel-openssl-1.0.x"]
}

generator erd {
  provider = "prisma-erd-generator-markdown"
  output   = "./ERD.md"
}

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

model User {
  id       String  @id
  username String?
  roomID   String
  room     Room    @relation(fields: [roomID], references: [id])

  GameAsPlayer1 Game[] @relation(name: "player_1")
  GameAsPlayer2 Game[] @relation(name: "player_2")

  @@index([roomID])
}

model Room {
  id    String @id
  users User[]
  game Game?
}

enum Status {
  PENDING
  ACTIVE
}

enum Turn {
  RED
  BLACK
}

model Game {
  id        String  @id
  roomID    String  @unique
  turn      Turn    @default(RED)
  room      Room    @relation(fields: [roomID], references: [id])
  status    Status  @default(PENDING)


  player1   User?   @relation(name: "player_1", fields: [player1ID], references: [id])
  player2   User?   @relation(name: "player_2", fields: [player2ID], references: [id])
  player1ID String?
  player2ID String?

  @@index([player1ID])
  @@index([player2ID])
}

```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by ludralph** [comment]
Hi @tranthaison1231 👋 

I think you are running into an issue similar to [this](https://github.com/prisma/prisma/issues/9520#issuecomment-1407261845), it's most probably caused by bundlers not copying the full generated Prisma Client to the new location and skipping the schema file or rewriting/bundling/moving the code of Prisma Client in a way that the schema file copy is not in the expected location either.

- https://github.com/prisma/prisma/issues/9520#issuecomment-1407261845

Could you try using [copy-webpack-plugin](https://www.npmjs.com/package/copy-webpack-plugin) to copy the `schema.prisma` file. It could look something similar to this:

```typescipt
const CopyWebpackPlugin = require('copy-webpack-plugin');

module.exports = {
  // other webpack config
  plugins: [
    new CopyWebpackPlugin({
      patterns: [
        { from: 'prisma/schema.prisma', to: 'prisma/schema.prisma' }
      ],
    }),
    // other plugins
  ]
};
``` 
This will tell webpack to copy the `schema.prisma` file into the same location in the output directory. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by thaisonenouvo** [reply]
Hi thank for your anwser @ludralph  But right now, i'm using esbuild on serverless, to do with prisma and pnpm workspace and turborepo, so do you have any idea for this

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
